snorkel/utils/optimizers.py

```diff
@@ -0,0 +1,34 @@
+from typing import Tuple
+
+from snorkel.types import Config
+
+
+class SGDOptimizerConfig(Config):
+    """Settings for SGD optimizer."""
+
+    momentum: float = 0.9
+
+
+class AdamOptimizerConfig(Config):
+    """Settings for Adam optimizer."""
+
+    amsgrad: bool = False
+    betas: Tuple[float, float] = (0.9, 0.999)
+
+
+class AdamaxOptimizerConfig(Config):
+    """Settings for Adamax optimizer."""
+
+    betas: Tuple[float, float] = (0.9, 0.999)
+    eps: float = 1e-8
+
+
+class OptimizerConfig(Config):
+    """Settings common to all optimizers."""
+
+    lr: float = 0.001
```

Comment:
Hmmmm this is a pretty critical value. Do we believe this is better than 0.01 for the label model? cc @paroma

Reply:
Can we surface lr and l2 one level higher? Just since they're particularly important

Reply:
Default `n_epochs` changed to account for lower LR. 0.01 is better for LabelModel (as in, it works most of the time with the different matrices we've had across tests and tutorials). I don't have a strong opinion, we can explicitly change LR in the label model if needed